src/ray/raylet/scheduling_resources.h

```diff
@@ -80,12 +80,14 @@ class ResourceSet {
   /// \return True, if the resource was successfully removed. False otherwise.
   bool RemoveResource(const std::string &resource_name);
 
-  /// \brief Add a set of resources to the current set of resources.
+  /// \brief Add a set of resources to the current set of resources (LEFT JOIN).
   ///
   /// \param other: The other resource set to add.
   /// \return True if the resource set was added successfully. False otherwise.
   bool AddResources(const ResourceSet &other);
 
+  void OuterJoin(const ResourceSet &other);
```

Comment:
Documentation? Also I personally didn't know what an outer join was. The implementation of this method looks very similar to `AddResources`, so it'd be nice to have naming that makes it clear that these two methods are related but different.

Reply:
Let's find a clearer name, I also don't know what an outer join is.